cloudera/stackstate_checks/cloudera/cloudera.py

```diff
@@ -0,0 +1,188 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+import json
+import logging
+import time
+
+import cm_client
+from cm_client.rest import ApiException
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+from stackstate_checks.base import AgentCheck, is_affirmative, TopologyInstance, ConfigurationError
+
+
+class ClouderaCheck(AgentCheck):
+    INSTANCE_TYPE = 'cloudera'
+    SERVICE_CHECK_NAME = 'cloudera.can_connect'
+    EVENT_TYPE = 'cloudera.entity_status'
+    EVENT_MESSAGE = '{} status'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self.url = None
+        self.tags = None
+        self.roles = None
+
+    def get_instance_key(self, instance):
+        if 'url' not in instance:
+            raise ConfigurationError('Missing url in topology instance configuration.')
+
+        instance_url = urlparse(instance['url']).netloc
+        return TopologyInstance(self.INSTANCE_TYPE, instance_url)
+
+    def check(self, instance):
+        self.url = get_config(instance)[0]
+
+        self.tags = ['instance_url: {}'.format(self.url)]
+        self.roles = []
+
+        try:
+            api_client = ClouderaClient(instance)
+
+            # collect topology
+            self.start_snapshot()
+            self._collect_topology(api_client)
+            self.stop_snapshot()
```

Comment:
`stop_snapshot` should always be executed otherwise in case of error we might send a snapshot without the stop and would take 2 check runs to recover.

but you don't want to stop if there was no start.

I would move start outside of the try and put stop in a finally clause

Reply:
Fixed :)